frontend/app/routes/public/renew/$id/ita/update-mailing-address.tsx

```diff
@@ -66,75 +60,43 @@ export async function loader({ context: { appContainer, session }, params, reque
 
 export async function action({ context: { appContainer, session }, params, request }: ActionFunctionArgs) {
   const formData = await request.formData();
+  const locale = getLocale(request);
 
   const securityHandler = appContainer.get(TYPES.routes.security.SecurityHandler);
   securityHandler.validateCsrfToken({ formData, session });
 
   const state = loadRenewItaState({ params, request, session });
-  const t = await getFixedT(request, handle.i18nNamespaces);
-  const { CANADA_COUNTRY_ID, USA_COUNTRY_ID } = appContainer.get(TYPES.configs.ClientConfig);
 
-  const mailingAddressSchema = z
-    .object({
-      address: z.string().trim().min(1, t('renew-ita:update-address.error-message.mailing-address.address-required')).max(30).refine(isAllValidInputCharacters, t('renew-ita:update-address.error-message.characters-valid')),
```

Comment:
Can we remove all these mailing address error messages from the i18n files if they're not being used?

Reply:
Be careful, the i18n locales used in `TYPES.routes.public.addressValidation.MailingAddressValidator` are specific to `addressValidation` route. We could move those validation error message within a common localization if they are the same everywhere.

https://github.com/DTS-STN/canadian-dental-care-plan/blob/7c3e3866124bf19d97a569033186634d8fddee01/frontend/app/.server/routes/public/address-validation/mailing-address.validator.ts#L30

Reply:
We shouldn't be referencing `MailingAddressValidator` at all in that case, since this validator is reserved for `address-validation` route.

We should use `AddressValidator` and specifying the error messages that are specific to this `/ita/update-mailing-address` route.

Reply:
Create a `MailingAddressValidator` and `MailingAddressValidatorFactory` under `./app/.server/routes/public/renew/ita/` that will uses `'renew-ita'` i18n namespace to provide error message to `AddressValidator`.

Reply:
Approving but I think we should consider having a single component that can validate address forms where we can pass the field names and error messages or similar. The current approach would lead to creating multiple validators for similar address forms, which could be avoided.